src/core/mount.c

```diff
@@ -1765,6 +1766,27 @@ static int mount_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
         Iterator i;
         Unit *u;
         int r;
+        uint64_t this_run = now(CLOCK_MONOTONIC);
+
+        if (source && this_run < m->mount_last_read + m->mount_last_duration * 10) {
+                /* If we run too often we can steal CPU from
+                 * the process that is mounting/unmounting things.
+                 */
+                sd_event_source_set_enabled(source, SD_EVENT_OFF);
```

Comment:
needs some error handling (even if it is just logging someithing at log_warning() level and then ignoring it